contracts/test/StakeLocker.t.sol

```diff
@@ -2,663 +2,254 @@
+    function test_intendToUnstake(uint256 amount) external {
+        amount = constrictToRange(amount, 1, type(uint256).max / block.timestamp);
 
-        sid.approve(address(bPool), address(stakeLocker1), stakeAmount);
-        sid.stake(address(stakeLocker1), stakeAmount);
+        stakeToken.mint(address(staker), amount);
+        staker.erc20_approve(address(stakeToken), address(locker), amount);
+        delegate.stakeLocker_openStakeLockerToPublic(address(locker));
+        staker.stakeLocker_stake(address(locker), amount);
 
-        // Staker 1 initiates unstake
-        assertTrue(sid.try_intendToUnstake(address(stakeLocker1)));
-        assertEq(stakeLocker1.unstakeCooldown(address(sid)), start);
+        assertTrue(staker.try_stakeLocker_intendToUnstake(address(locker)));
+        
+        assertEq(locker.unstakeCooldown(address(staker)), block.timestamp);
+    }
 
-        // Staker 2 fails to transfer to Staker 1 that is currently unstaking
-        assertTrue(!sam.try_transfer(address(stakeLocker1), address(sid), stakeAmount));
-        hevm.warp(start + globals.stakerCooldownPeriod() + globals.stakerUnstakeWindow());  // Very end of Staker unstake window
-        assertTrue(!sam.try_transfer(address(stakeLocker1), address(sid), stakeAmount));
+    function test_cancelUnstake(uint256 amount) external {
+        amount = constrictToRange(amount, 1, type(uint256).max / block.timestamp);
 
-        // Staker 2 successfully transfers to Staker 1 that is now outside unstake window
-        hevm.warp(start + globals.stakerCooldownPeriod() + globals.stakerUnstakeWindow() + 1);  // Second after Staker unstake window ends
-        uint256 newStakeDate = getNewStakeDate(address(sid), stakeAmount);
-        assertTrue(sam.try_transfer(address(stakeLocker1), address(sid), stakeAmount));
+        stakeToken.mint(address(staker), amount);
+        staker.erc20_approve(address(stakeToken), address(locker), amount);
+        delegate.stakeLocker_openStakeLockerToPublic(address(locker));
+        staker.stakeLocker_stake(address(locker), amount);
+        staker.stakeLocker_intendToUnstake(address(locker));
 
-        // Check balances and stake dates are correct
-        assertEq(stakeLocker1.balanceOf(address(sam)), 0);
-        assertEq(stakeLocker1.balanceOf(address(sid)), stakeAmount * 2);
-        assertEq(stakeLocker1.stakeDate(address(sam)), start);         // Stays the same
-        assertEq(stakeLocker1.stakeDate(address(sid)), newStakeDate);  // Gets updated
+        assertTrue(staker.try_stakeLocker_cancelUnstake(address(locker)));
+        
+        assertEq(locker.unstakeCooldown(address(staker)), 0);
     }
 
-    function setUpLoanAndRepay() public {
-        mint("USDC", address(leo), 10_000_000 * USD);  // Mint USDC to LP
-        leo.approve(USDC, address(pool1), MAX_UINT);   // LP approves USDC
+    function test_unstake(uint256 stakeAmount, uint256 unstakeAmount) external {
```

Comment:
You can ignore the below comments, here is a full test:
```js
      function test_unstake(uint256 stakeAmount, uint256 unstakeAmount, uint256 unstakeAmount2) external {
        stakeAmount    = constrictToRange(stakeAmount,    2, type(uint256).max / block.timestamp, true);  // Min 2 so two unstakes can be done
        unstakeAmount  = constrictToRange(unstakeAmount,  1, stakeAmount,                         true);
        unstakeAmount2 = constrictToRange(unstakeAmount2, 1, stakeAmount - unstakeAmount,         true);

        stakeToken.mint(address(staker), stakeAmount);
        staker.erc20_approve(address(stakeToken), address(locker), stakeAmount);
        delegate.stakeLocker_openStakeLockerToPublic(address(locker));
        staker.stakeLocker_stake(address(locker), stakeAmount);
        staker.stakeLocker_intendToUnstake(address(locker));

        uint256 start = block.timestamp;

        hevm.warp(start + locker.lockupPeriod());

        // Can't unstake immediately (lockup not expired)
        assertTrue(!staker.try_stakeLocker_unstake(address(locker), unstakeAmount));

        staker.stakeLocker_cancelUnstake(address(locker));

        staker.stakeLocker_intendToUnstake(address(locker));
        hevm.warp(start + locker.lockupPeriod() + globals.stakerCooldownPeriod() - 1);

        // Can't unstake before cooldown is over
        assertTrue(!staker.try_stakeLocker_unstake(address(locker), unstakeAmount));

        hevm.warp(start + locker.lockupPeriod() + globals.stakerCooldownPeriod() + globals.stakerUnstakeWindow() + 1);

        // Can't unstake past unstake window
        assertTrue(!staker.try_stakeLocker_unstake(address(locker), unstakeAmount));

        hevm.warp(start + locker.lockupPeriod() + globals.stakerCooldownPeriod() + globals.stakerUnstakeWindow());

        // Can unstake at end of unstake window
        assertTrue(staker.try_stakeLocker_unstake(address(locker), unstakeAmount));

        hevm.warp(start + locker.lockupPeriod());
        staker.stakeLocker_intendToUnstake(address(locker));
        hevm.warp(start + locker.lockupPeriod() + globals.stakerCooldownPeriod());

        // Can unstake at beginning of cooldown
        assertTrue(staker.try_stakeLocker_unstake(address(locker), unstakeAmount2));

        assertEq(locker.balanceOf(address(staker)), stakeAmount - unstakeAmount - unstakeAmount2);
    }

Reply:
I updated it in a different way that tests more explicit cases.